proxy/coco/src/state.rs

```diff
@@ -0,0 +1,733 @@
+        let (api, _run_loop) = config.try_into_peer().await?.accept()?;
+        let state = State::new(api, signer.clone());
+
+        let cloudhead = state.init_user(&signer, "cloudhead")?;
+        let _cloudhead = super::verify_user(cloudhead)?;
+        let kalt = state.init_user(&signer, "kalt")?;
+        let _kalt = super::verify_user(kalt)?;
+
+        let users = state.list_users()?;
+        let mut user_handles = users
+            .into_iter()
+            .map(|user| user.name().to_string())
+            .collect::<Vec<_>>();
+        user_handles.sort();
+
+        assert_eq!(user_handles, vec!["cloudhead", "kalt"],);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn create_with_existing_remote_with_reset() -> Result<(), Error> {
+        let tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+        let repo_path = tmp_dir.path().join("radicle");
+        let key = SecretKey::new();
+        let signer = signer::BoxedSigner::from(key.clone());
+        let config = config::default(key, tmp_dir.path())?;
+        let (api, _run_loop) = config.try_into_peer().await?.accept()?;
+        let state = State::new(api, signer.clone());
+
+        let kalt = state.init_owner(&signer, "kalt")?;
+
+        let fakie = state.init_project(&signer, &kalt, &fakie_project(repo_path.clone()))?;
+
+        assert!(repo_path.join(fakie.name()).exists());
+
+        // Simulate resetting the monorepo
+        let tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+        let key = SecretKey::new();
+        let signer = signer::BoxedSigner::from(key.clone());
+        let config = config::default(key, tmp_dir.path())?;
+        let (api, _run_loop) = config.try_into_peer().await?.accept()?;
+        let state = State::new(api, signer.clone());
+
+        // Create fakie project from the existing directory above.
+        let kalt = state.init_owner(&signer, "kalt")?;
+        let fakie =
+            state.init_project(&signer, &kalt, &fakie_project(repo_path).into_existing())?;
+
+        // Attempt to initialise a browser to ensure we can look at branches in the project
+        let _stats = state.with_browser(&fakie.urn(), |browser| Ok(browser.get_stats()?))?;
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn create_with_existing_remote() -> Result<(), Error> {
+        let tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+        let repo_path = tmp_dir.path().join("radicle");
+        let key = SecretKey::new();
+        let signer = signer::BoxedSigner::from(key.clone());
+        let config = config::default(key, tmp_dir.path())?;
+        let (api, _run_loop) = config.try_into_peer().await?.accept()?;
+        let state = State::new(api, signer.clone());
+
+        let kalt = state.init_owner(&signer, "kalt")?;
+
+        let fakie = state.init_project(&signer, &kalt, &fakie_project(repo_path.clone()))?;
+
+        let fake_fakie = repo_path.join("fake-fakie");
```

Comment:
Though I enjoy some humour, I got no idea what `fake_fakie` means to represent without digesting the whole block.

Reply:
@FintanH knows the fakies.

Reply:
Do you have a suggestion here?

We know that it's a `PathBuf` because of `repo_path.join` and it's used as a target directory for `cp`. The whole context is the test itself.

`copied_fakie` maybe? :shrug: